pygeoapi/provider/base_edr.py

```diff
@@ -95,3 +95,10 @@ def query(self, **kwargs):
             return getattr(self, kwargs.get('query_type'))(**kwargs)
         except AttributeError:
             raise NotImplementedError('Query not implemented!')
+
+
+class EdrProviderRequestEntityTooLarge(ProviderGenericError):
+    """provider request entity too large error"""
+
+    def __init__(self, request_limit_explanation: str):
+        self.request_limit_explanation = request_limit_explanation
```

Comment:
Do we need `request_limit_explanation`?  At runtime we would be able to get the same error message via:

```python
if 1 != 2:
    raise ProviderRequestEntityTooLargeError('Requested area too large')
```